components/src/dynamo/common/tests/test_guided_json.py

```
@@ -0,0 +1,157 @@
+        "serialized-root-self-reference",
+        "root-self-reference-with-sibling",
+        "two-definition-cycle",
+        "three-definition-cycle",
+        "cycle-through-all-of",
+        "unused-definition-cycle",
+    ],
+)
+def test_rejects_unproductive_local_reference_cycles(schema):
+    with pytest.raises(
+        HttpError,
+        match="unproductive circular.*ref",
```

Ruff (0.16.1)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)